tiatoolbox/utils/env_detection.py

```diff
@@ -23,13 +24,15 @@
 import subprocess
 import sys
 import threading
-from numbers import Number
-from typing import List, Tuple
+from typing import TYPE_CHECKING
 
 import torch
 
 from tiatoolbox import logger
 
+if TYPE_CHECKING:  # pragma: no cover
+    from numbers import Number
```

Comment:
We might want to swap from Number to a custom type e.g. Union[int, float]. See this thread: https://stackoverflow.com/questions/60616802/how-to-type-hint-a-generic-numeric-type-in-python, which suggests using:
```python
from typing import SupportsFloat as Numeric
```
Some funky things can happen with Number e.g. because it includes complex numbers.

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/pull/625/commits/bb24fccd10322177197c25b2192531eb94e9a6b4